code/modules/mob/mob_movement.dm

```diff
@@ -302,7 +302,7 @@
 
 		//We are now going to move
 		move_delay = max(move_delay,1)
-		if(mob.movement_speed_modifier)
+		if(mob.movement_speed_modifier && mob.movement_speed_modifier != 1)
```

Comment:
A mathematical operation is still less costly than a logical operation if I remember correctly, so you might as well let it multiply